src/theory/quantifiers/sygus/sygus_pbe.h

```diff
@@ -361,40 +391,42 @@ class CegConjecturePbe : public SygusModule
     /** the children nodes of this trie */
     std::map<Node, SubsumeTrie> d_children;
     /** helper function for above functions */
-    Node addTermInternal(CegConjecturePbe* pbe,
-                         Node t,
+    Node addTermInternal(Node t,
                          std::vector<Node>& vals,
                          bool pol,
                          std::vector<Node>& subsumed,
                          bool spol,
-                         IndexFilter* f,
                          unsigned index,
                          int status,
                          bool checkExistsOnly,
                          bool checkSubsume);
     /** helper function for above functions */
-    void getLeavesInternal(CegConjecturePbe* pbe,
-                           std::vector<Node>& vals,
+    void getLeavesInternal(std::vector<Node>& vals,
                            bool pol,
                            std::map<int, std::vector<Node> >& v,
-                           IndexFilter* f,
                            unsigned index,
                            int status);
   };
   // -------------------------------- end decision tree learning
 
   //------------------------------ representation of a enumeration strategy
-
-  /** information about an enumerator
-   *
-   * We say an enumerator is a master enumerator if it is the variable that
-   * we use to enumerate values for its sort. Master enumerators may have
-   * (possibly multiple) slave enumerators, stored in d_enum_slave,
-   */
-  class EnumInfo {
+  /**
+  * This class stores information regarding an enumerator, including:
+  * - Information regarding the role of this enumerator (see EnumRole), its
+  * parent, whether it is templated, its slave enumerators, and so on, and
+  * - A database of values that have been enumerated for this enumerator.
+  *
+  * We say an enumerator is a master enumerator if it is the variable that
```

Comment:
clarify difference between master and slave enumerator, as in, in which situations is one used rather than the other?